src/Types/ContractName.cs

```diff
@@ -1,6 +1,90 @@
 namespace Concordium.Sdk.Types;
 
 /// <summary>
-/// A contract name. Expected format: "init_&lt;contract_name&gt;".
+/// The init name of a smart contract function. Expected format:
+/// "init_&lt;contract_name&gt;". It must only consist of at most 100 ASCII
+/// alphanumeric or punctuation characters, must not contain a '.' and must
+/// start with 'init_'.
 /// </summary>
-public sealed record ContractName(string Contract);
+public sealed record ContractName
+{
+    private const uint MaxByteLength = 100;
+
+    /// <summary>
+    /// A contract name with format: "init_&lt;contract_name&gt;".
+    /// </summary>
+    public string Name { get; init; }
+
+    private ContractName(string name) => this.Name = name;
+
+    internal static ContractName From(Grpc.V2.InitName initName) => new(initName.Value);
+
+    /// <summary>
+    /// Try parse input name against expected format.
+    /// </summary>
+    /// <param name="name">Input init name.</param>
+    /// <param name="initName">Parsed init name when succeeded.</param>
+    /// <returns>True if name satisfied expected format.</returns>
+    public static bool TryParse(string name, out ContractName? initName)
+    {
+        var validate = IsValid(name);
+        initName = validate ? new ContractName(name) : null;
+        return validate;
+    }
+
+    private static bool IsValid(string name)
+    {
+        if (!name.StartsWith("init_", StringComparison.Ordinal))
+        {
+            return false;
+        }
+        if (name.Length > MaxByteLength)
+        {
+            return false;
+        }
+        if (name.Contains('.'))
+        {
+            return false;
+        }
+        if (!name.All(c => char.IsLetterOrDigit(c) || IsAsciiPunctuation(c)))
```

Comment:
This will accept unicode letters as well and we only allow ascii. This seems more fitting
https://learn.microsoft.com/en-us/dotnet/api/system.char.isasciiletter?view=net-7.0
```suggestion
        if (!name.All(c => char.IsAsciiLetter(c) || IsAsciiPunctuation(c)))
```

Reply:
The suggestion uses methods from net 7 and we use net 6.

I have made own helper method.